src/lib/annotations/drawing/DrawingThread.js

```diff
@@ -0,0 +1,169 @@
+/* global Rbush */
+import AnnotationThread from '../AnnotationThread';
+import { DRAW_POINTER_UP, DRAW_RENDER_THRESHOLD } from '../annotationConstants';
+
+const RTREE_WIDTH = 5; // Lower number - faster search, higher - faster insert
+const BASE_LINE_WIDTH = 3;
+
+class DrawingThread extends AnnotationThread {
+    /** @property {number} - Drawing state */
+    drawingFlag = DRAW_POINTER_UP;
+
+    /** @property {Rbush} - Rtree path container */
+    pathContainer = new Rbush(RTREE_WIDTH);
+
+    /** @property {CanvasContext} - A canvas for drawing new strokes */
+    memoryCanvas;
+
+    /** @property {DrawingPath} - The path being drawn but not yet finalized */
+    pendingPath;
+
+    /** @property {CanvasContext} - The context to be drawn on */
+    context;
+
+    /** @property {number} - Timestamp of the last render */
+    lastRenderTimestamp;
+
+    /** @property {number} - The the last animation frame request id */
+    lastAnimRequestId;
+
+    /**
+     * [constructor]
+     *
+     * @inheritdoc
+     * @param {AnnotationThreadData} data - Data for constructing thread
+     * @return {DrawingThread} Drawing annotation thread instance
+     */
+    constructor(data) {
+        super(data);
+        this.render = this.render.bind(this);
+    }
+
+    /**
+     * Soft destructor for a drawingthread object
+     *
+     * [destructor]
+     * @inheritdoc
+     * @return {void}
+     */
+    destroy() {
+        if (this.lastAnimRequestId) {
+            window.cancelAnimationFrame(this.lastAnimRequestId);
+        }
+        this.removeAllListeners();
+        this.reset();
+        super.destroy();
+    }
+
+    /**
+     * Get all of the DrawingPaths in the current thread.
+     *
+     * @return {void}
+     */
+    getDrawings() {
+        return this.pathContainer.all();
+    }
+
+    /* eslint-disable no-unused-vars */
+    /**
+     * Handle a pointer movement
+     *
+     * @param {Object} location - The location information of the pointer
+     * @return {void}
+     */
+    handleMove(location) {}
+
+    /**
+     * Start a drawing stroke *
+     *
+     * @param {Object} location - The location information of the pointer
+     * @return {void}
+     */
+    handleStart(location) {}
+
+    /**
+     * End a drawing stroke
+     *
+     * @param {Object} location - The location information of the pointer
+     * @return {void}
+     */
+    handleStop(location) {}
+    /* eslint-disable no-unused-vars */
+
+    //--------------------------------------------------------------------------
+    // Protected
+    //--------------------------------------------------------------------------
+
+    /**
+     * Set the drawing styles
+     *
+     * @protected
+     * @param {number} scale - The document scale
+     * @return {void}
+     */
+    setContextStyles(scale) {
```

Comment:
I think we should pass in a config object here, so that it'll be easier when users ask for colours, line width, etc.